activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -559,7 +559,7 @@ def find_nth_from_last(index)
         else
           relation = ordered_relation
 
-          if equal?(relation) || has_limit_or_offset?
+          if relation.order_values.empty? || has_limit_or_offset?
```

Comment:
Digging into `ordered_relation`, I can see how this works, but it reads a bit strangely to me.  Should we add `relation.` on both conditions?

```suggestion
          if relation.order_values.empty? || relation.has_limit_or_offset?
```

Also, trying to improve my understanding of this code, I think `relation.has_limit_or_offset?` is not necessary for correctness, because [`last` performs the same check](https://github.com/rails/rails/blob/01f58d62c2f31f42d0184e0add2b6aa710513695/activerecord/lib/active_record/relation/finder_methods.rb#L172).  But it does save an array allocation, because `relation.last(index)[-index]` is equivalent to `relation.records.last(index)[-index]` when `relation.has_limit_or_offset?` is true.  Does that sound correct to you?

Reply:
Yes, makes sense. I rewrote it slightly to not retrieve unneeded extra records.

Reply:
Nice! :rocket: